storage/innobase/trx/trx0roll.cc

```diff
@@ -197,6 +200,12 @@ static dberr_t trx_rollback_low(trx_t *trx) {
     case TRX_STATE_NOT_STARTED:
       trx->will_lock = 0;
       ut_ad(trx->in_mysql_trx_list);
+#ifdef WITH_WSREP
+      /* If the transaction is aborted, clean this flag.
+      Otherwise, a new transaction will use this trx
+      object and will be treated as a victim */
+      trx->lock.was_chosen_as_wsrep_victim = false;
```

Comment:
The flow is no so obvious here....
Maybe worth mentioning that killing transaction calls rollback() which calls  commit(), which detects that transaction was aborted and sets its state to TRX_STATE_FORCED_ROLLBACK. 
Then, when rollback is called again after commit prepare failure, we are detecting this flag here.

Reply:
updated comment